### ISSUE TITLE
[Text][Accessibility] textDecorationLine="line-through" does not announce itself on a screen reader

### ISSUE THREAD

**Issue by stephxshopify** [Shopify/polaris-react-archive#12451]
### Summary

VoiceOver does not read strike throughs.
This can be confusing to screen readers when prices are shown with a strikethrough old price and a new price displayed beside it.

### Expected behavior

Screen reader should read text decorations

### Actual behavior

<img width="859" alt="Screenshot 2024-07-24 at 3 56 37 PM" src="https://github.com/user-attachments/assets/f737704b-df64-4b88-ba6a-92856cc4ca91">

### Steps to reproduce

https://codesandbox.io/p/sandbox/dry-tree-9ml5c5?file=%2FApp.tsx%3A12%2C5

Pass over text with textDecorationLine="line-through" and view screen reader does not note there are any text decorations

### Are you using React components?

None

### Browser

Chrome

### Device

MacBook Pro
